fictadvisor-api/src/v2/google/services/GoogleCalendarService.ts

```diff
@@ -0,0 +1,451 @@
+import { GoogleAuthService } from './GoogleAuthService';
+import { GoogleCalendarAPI, KYIV_TIMEZONE } from '../apis/GoogleCalendarAPI';
+import { NoGoogleGrantException } from '../../utils/exceptions/NoGoogleGrantException';
+import { GoogleUserRepository } from '../../database/repositories/GoogleUserRepository';
+import { EventRepository } from '../../database/repositories/EventRepository';
+import { DateService } from '../../utils/date/DateService';
+import { GoogleNotLinkedException } from '../../utils/exceptions/GoogleNotLinkedException';
+import { UserRepository } from '../../database/repositories/UserRepository';
+import { InvalidEntityIdException } from '../../utils/exceptions/InvalidEntityIdException';
+import { DbEvent } from '../../database/entities/DbEvent';
+import { CreateGoogleEventDTO } from '../types/calendar/CreateGoogleEventDTO';
+import { EventInstancesResponse } from '../types/calendar/EventInstancesResponse';
+import { DisciplineRepository } from '../../database/repositories/DisciplineRepository';
+
+
+const EventColorAdapter: Record<EventTypeEnum, string> = {
+  [EventTypeEnum.LECTURE]: '9',
+  [EventTypeEnum.PRACTICE]: '6',
+  [EventTypeEnum.LABORATORY]: '2',
+  [EventTypeEnum.CONSULTATION]: '10',
+  [EventTypeEnum.EXAM]: '3',
+  [EventTypeEnum.WORKOUT]: '4',
+  [EventTypeEnum.OTHER]: '8',
+};
+
+
+@Injectable()
+export class GoogleCalendarService {
+  constructor (
+    private config: GoogleConfigService,
+    private googleAuthService: GoogleAuthService,
+    private dateService: DateService,
+    private calendarAPI: GoogleCalendarAPI,
+    private googleUserRepository: GoogleUserRepository,
+    private eventRepository: EventRepository,
+    private userRepository: UserRepository,
+    private disciplineRepository: DisciplineRepository,
+  ) {}
+
+  async checkUserCalendar(googleId: string): Promise<boolean> {
+    if (!googleId)
+      throw new GoogleNotLinkedException();
+
+    const accessToken = await this.googleAuthService.getAccessToken(googleId);
+    if (!accessToken || !(await this.googleAuthService.hasCalendarPermissions(accessToken)))
+      throw new NoGoogleGrantException();
+
+    const { calendarId } = await this.googleUserRepository.findById(googleId);
+    if (!calendarId) {
+      return false;
+    }
+
+    return await this.hasCalendar(googleId, calendarId, accessToken);
+  }
+
+  private async hasCalendar(googleId: string, calendarId: string, accessToken: string): Promise<boolean> {
+    try {
+      await this.calendarAPI.getCalendarFromUserList(calendarId, accessToken);
+    } catch (e) {
+      await this.googleUserRepository.updateById(googleId, {
+        calendarId: null,
+      });
+      await this.deleteCalendar(calendarId, accessToken);
+
+      return false;
+    }
+
+    return true;
+  }
+
+  private async deleteCalendar(calendarId: string, accessToken: string): Promise<void> {
+    try {
+      await this.calendarAPI.deleteCalendar(calendarId, accessToken);
+    } catch (e) {}
```

Comment:
In my opinion, such an exception would make no sense here, as this exception is not a request error on the client side, and the only purpose of this try catch block is not to intrude into the flow of other asynchronous tasks. This function is designated to delete the user's calendar in case it's been deleted from the user's calendar list (which makes it unreachable for anyone), and since it could have already been deleted by the user manually, the handling of this exception is unnecessary